### PR TITLE
Detection of cmake output correctly

### PR DESCRIPTION
**Pull request by Mintyboi**
This is because we are checking if the output of the cmake contains "Emscripten" (case sensitive), which does not exists when we skip the sanity check.
The word "Emscripten" is part of an output log inside check_sanity() function, `logging.info('(Emscripten: Running sanity checks)')`

By ignoring the case of the initial character, I hope to detect the cmake output correctly

### CODE REVIEW

**Comment by kripken** [comment]
@juj, can you look at this one?

**Comment by juj** [comment]
Oh, very good catch! The CMake detection code runs `emcc -v`, which outputs

	clb@x10dax ~/emsdk/emscripten/incoming/tests/cmake/target_html/t $ emcc -v
	emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 1.37.12
	clang version 4.0.0 (https://github.com/kripken/emscripten-fastcomp-clang/ ac7f972a03d729dddd614882b634da3884962263) (https://github.com/kripken/emscripten-fastcomp/ 042f4a81b1d95123ef6798ffc1457da3217d44bc) (emscripten 1.37.12 : 1.37.12)
	Target: x86_64-unknown-linux-gnu
	Thread model: posix
	InstalledDir: /home/clb/emsdk/clang/fastcomp/build_incoming_64/bin
	Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/5
	Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/5.4.0
	Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/6
	Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/6.0.0
	Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/5
	Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/5.4.0
	Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6
	Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6.0.0
	Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/5.4.0
	Candidate multilib: .;@m64
	Selected multilib: .;@m64
	INFO:root:(Emscripten: Running sanity checks)

The last line is not present if `EMCC_SKIP_SANITY_CHECK=1`, but the first line is always present (result from `clang -v` which `emcc` itself calls), so always looking for the lowercase string works. Pushed https://github.com/kripken/emscripten/commit/909e396f567cf52b556b8d03a4e6380f732aa1d3 to fix this and add test coverage as well.

(oh also, pull requests are landed against `incoming` branch)

Thanks for noticing this!

**Comment by Mintyboi** [comment]
Thank you for making the change.
Sorry for the request against the wrong branch! I'll take note in the future.

File: cmake/Modules/Platform/Emscripten.cmake [hunk 1]
```diff
@@ -107,7 +107,7 @@ if (EMSCRIPTEN_FORCE_COMPILERS)
 		if (NOT _cmake_compiler_result EQUAL 0)
 			message(FATAL_ERROR "Failed to fetch compiler version information with command \"'${CMAKE_C_COMPILER}' -v\"! Process returned with error code ${_cmake_compiler_result}.")
 		endif()
-		if (NOT "${_cmake_compiler_output}" MATCHES "Emscripten")
+		if (NOT "${_cmake_compiler_output}" MATCHES "(E|e?)mscripten")
```
Comment (jm4R) [inline cmake/Modules/Platform/Emscripten.cmake:110]: What about `~/emsdk/upstream/bin/clang` path? Isn't it common?

**Pull request closed without merging**